core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -304,7 +304,7 @@ impl<Tx: TransactionWithMeta> Scheduler<Tx> for PrioGraphScheduler<Tx> {
 
         // Push remaining transactions back into the container
         while let Some((id, _)) = self.prio_graph.pop_and_unblock() {
-            container.push_id_into_queue(id);
+            container.put_id_back_into_queue(id);
```

Comment:
A separate note: during debugging, I noticed those IDs being put back are earliest txs, eg `{priority: 1; id: 0}`, `{priority: 1; id: 1}`... I would think given same priority, txs received earlier should be scheduled earlier. This doesn't seem the case tho.

Reply:
ids are recycled in a stack because that's how `Slab` finds them; it's an index

Reply:
I meant the `ids that are `pop_and_unblock()` from prio_graph, not before being reinserted into Container.